### ISSUE TITLE
about the loss in train_binclass.py

### ISSUE THREAD

**Issue by algorithmlixuan** [polimi-ispl/icpr2020dfdc#16]
Hi, thanks for the great work. I have some questions about your loss in train_binclass.py. In your paper, I feel the input of your BCEloss are labels and sigmoid(out) ,but in your code, the input of the BCEloss are out and labels.Would you like to tell me why you write your BCEloss code like that? thank you very much
![image](https://user-images.githubusercontent.com/30352320/110442139-86181480-80f5-11eb-9d1b-983d3425ab19.png)

![image](https://user-images.githubusercontent.com/30352320/110442107-7b5d7f80-80f5-11eb-8ef0-3a1f15170a8a.png)

**Comment by CrohnEngineer** [comment]
Hey @algorithmlixuan ,

> Hi, thanks for the great work.

thank you for checking out!

> Would you like to tell me why you write your BCEloss code like that? thank you very much

The reason is that we use as `criterion` the [BCEWithLogitsLoss](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html#torch.nn.BCEWithLogitsLoss), which computes for us the sigmoid and then the BCE: therefore, we can pass to the function directly the output `out` of the network and not the vector `pred` of predictions.
You can alternatively use as `criterion = BCELoss()` [(BCELoss](https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html#torch.nn.BCELoss)) and then compute `loss = criterion(pred, labels)`, passing in this case to the loss function the `pred` vectors.
Hope this answers your question!
Cheers :)

Edoardo
